src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CollectInsertSizeMetricsSpark.java

```diff
@@ -219,30 +226,29 @@ public SVCustomReadFilter(final UseEnd whichEnd,
                                   final boolean filterDupReads,
                                   final boolean filterSecondaryAlignments,
                                   final boolean filterSupplementaryAlignments,
-                                  final int MQThreshold){
+                                  final int MQThreshold,
+                                  final int mMQThreshold){
 
-                final UseEnd endVal = whichEnd;
+            final UseEnd endVal = whichEnd;
 
-                ReadFilter tempFilter = read -> 0!=read.getFragmentLength();
-                           tempFilter = tempFilter.and(read -> read.isPaired());
-                           tempFilter = tempFilter.and(read -> endVal == (read.isFirstOfPair() ? UseEnd.FIRST : UseEnd.SECOND));
+            ReadFilter tempFilter = read -> 0!=read.getFragmentLength();
+                       tempFilter = tempFilter.and(read -> read.isPaired());
+                       tempFilter = tempFilter.and(read -> endVal == (read.isFirstOfPair() ? UseEnd.FIRST : UseEnd.SECOND));
 
-                if(filterNonProperlyPairdReads)   { tempFilter = tempFilter.and(read -> read.isProperlyPaired());}
-                if(filterUnmappedReads)           { tempFilter = tempFilter.and(read -> !read.isUnmapped());}
-                if(filterMateUnmappedReads)       { tempFilter = tempFilter.and(read -> !read.mateIsUnmapped());}
-                if(filterDupReads)                { tempFilter = tempFilter.and(read -> !read.isDuplicate());}
-                if(filterSecondaryAlignments)     { tempFilter = tempFilter.and(read -> !read.isSecondaryAlignment());}
-                if(filterSupplementaryAlignments) { tempFilter = tempFilter.and(read -> !read.isSupplementaryAlignment());}
+            if(filterNonProperlyPairdReads)   { tempFilter = tempFilter.and(read -> read.isProperlyPaired());}
+            if(filterUnmappedReads)           { tempFilter = tempFilter.and(read -> !read.isUnmapped());}
+            if(filterMateUnmappedReads)       { tempFilter = tempFilter.and(read -> !read.mateIsUnmapped());}
+            if(filterDupReads)                { tempFilter = tempFilter.and(read -> !read.isDuplicate());}
+            if(filterSecondaryAlignments)     { tempFilter = tempFilter.and(read -> !read.isSecondaryAlignment());}
+            if(filterSupplementaryAlignments) { tempFilter = tempFilter.and(read -> !read.isSupplementaryAlignment());}
 
-                if(0!=MQThreshold){ tempFilter = tempFilter.and(read -> read.getMappingQuality() >= MQThreshold);}
+            if(0!=MQThreshold)  { tempFilter = tempFilter.and(read -> read.getMappingQuality() >= MQThreshold);}
+            if(0!=mMQThreshold) { tempFilter = tempFilter.and(read -> read.getAttributeAsInteger("MQ") >= mMQThreshold);}
```

Comment:
I think this would blow up if you ran this with filterMateUnmappedReads == false (in which case you'd get reads with unmapped mates which don't have an MQ tag I think). As I said above you'd never want to do that in this tool so I'd recommend removing the option but if you want to keep it in a reusable filter class I think you need a null check here.